### PR TITLE
Forecast job refactoring

### PR DESCRIPTION
**Pull request by sserurich**
**_WHAT DOES THIS PR DO?_**
- adds authentication header to api calls 
**_WHAT ISSUES ARE RELATED TO THIS PR?_**

**_HOW DO I TEST OUT THIS PR?_**
   - Fetch this branch and Checkout the branch locally
   - Change directory to Predict/jobs/train/
   - Create a python virtual environment (python3 -m venv env)
   - Install the necessary packages (pip install -r requirements.txt). 
   - Set the environment in the .env file to staging and run python main.py.

- If you have docker, simply run docker build -t <image_name> --target=staging .. Wait for the build process to complete and then run docker run -it <image_name>

- NB:
- .env and service account key is available from the secret manager.
- can check whether job has ran successfuly by observing the results on console and checking the model tracking mlflow interface 
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @sserurich 
LGTM

File: src/predict/jobs/train/models.py [hunk 1]
```diff
@@ -77,15 +88,18 @@ def get_events(self):
...
     def get_devices():
-  
+
         try:
-            api_url = f'{configuration.AIRQO_API_BASE_URL}devices?tenant={configuration.TENANT}&active=yes'
-            results = requests.get(api_url,verify=False)
+            api_url = f"{configuration.AIRQO_API_BASE_URL}devices?tenant={configuration.TENANT}&active=yes&network={configuration.TENANT}"
+            authorisation_token = f"'TOK:<{configuration.AIRQO_API_AUTH_TOKEN}>'"
+            results = requests.get(
+                api_url, verify=False, headers={"Authorization": authorisation_token}
```
Comment (Baalmart) [inline src/predict/jobs/train/models.py:98]: I hope this token has the JWT string at the beginning? Thanks @sserurich  and @123MwanjeMike

**Review by Baalmart — COMMENTED** [review]
All looks good, thanks @sserurich , I have just left some few comments in the PR. cc: @123MwanjeMike

File: src/predict/jobs/train/models.py (hunk 1 above)
Comment (123MwanjeMike) [reply src/predict/jobs/train/models.py:98]: Hi @sserurich 
We can now authenticate using access tokens as query parameters without using the JWT bearer tokens. Closing the PR...

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
closing PR...

**Comment by 123MwanjeMike** [comment]
obsolete...

**Pull request closed without merging**